spec/prog/postgres/converge_postgres_resource_spec.rb

```diff
@@ -3,403 +3,484 @@
+        version: "17",
+        representative_at: Time.now,
+        vm: instance_double(Vm, vm_host: instance_double(VmHost), nic: instance_double(Nic, nic_aws_resource: instance_double(NicAwsResource, subnet_az: "a")))
+      )
+      servers = [representative_server]
+      allow(postgres_resource.location).to receive(:provider).and_return(HostProvider::AWS_PROVIDER_NAME)
+      allow(postgres_resource).to receive_messages(servers: servers, has_enough_fresh_servers?: false, representative_server: representative_server, use_different_az_set?: false)
+
       expect(Prog::Postgres::PostgresServerNexus).to receive(:assemble).with(hash_including(availability_zone: "a"))
-      expect(postgres_resource).to receive(:use_different_az_set?).and_return(false)
       expect { nx.provision_servers }.to nap
     end
 
     it "provisions a new server with the correct timeline for a regular instance" do
       allow(Config).to receive(:allow_unspread_servers).and_return(true)
-      allow(postgres_resource).to receive(:read_replica?).and_return(false)
-      expect(postgres_resource.location).to receive(:provider).and_return(HostProvider::HETZNER_PROVIDER_NAME).at_least(:once)
-
-      expect(Prog::Postgres::PostgresServerNexus).to receive(:assemble).with(hash_including(timeline_id: "timeline-id"))
+      expect(Prog::Postgres::PostgresServerNexus).to receive(:assemble).with(hash_including(timeline_id: timeline.id))
       expect { nx.provision_servers }.to nap
     end
 
     it "provisions a new server with the correct timeline for a read replica" do
+      parent_timeline = PostgresTimeline.create(location_id: location_id)
+      parent = PostgresResource.create(
+        name: "pg-parent3", superuser_password: "dummy-password", ha_type: "none",
+        target_version: "17", location_id: location_id, project_id: project.id,
+        user_config: {}, pgbouncer_user_config: {}, target_vm_size: "standard-2",
+        target_storage_size_gib: 64, private_subnet_id: private_subnet.id
+      )
+      Strand.create_with_id(parent, prog: "Postgres::PostgresResourceNexus", label: "wait")
+      # Create a server for parent so it has a timeline
+      create_server(timeline: parent_timeline, representative: true, timeline_access: "push", resource: parent)
+
+      postgres_resource.update(parent_id: parent.id)
       allow(Config).to receive(:allow_unspread_servers).and_return(true)
-      allow(postgres_resource).to receive_messages(read_replica?: true, parent: instance_double(PostgresResource, timeline: instance_double(PostgresTimeline, id: "rr-timeline-id")))
-      expect(postgres_resource.location).to receive(:provider).and_return(HostProvider::HETZNER_PROVIDER_NAME).at_least(:once)
 
-      expect(Prog::Postgres::PostgresServerNexus).to receive(:assemble).with(hash_including(timeline_id: "rr-timeline-id"))
+      expect(Prog::Postgres::PostgresServerNexus).to receive(:assemble).with(hash_including(timeline_id: parent_timeline.id))
```

Comment:
Better to check after the method call that the created PostgresServer is using that timeline.